### ISSUE TITLE
Puzzles with syntax error instead of wrong answer

### ISSUE THREAD

**Issue by kokosek** [Code-Fights/codefights-wiki#71]
reflectString

**Comment by tigransloyan** [comment]
@Albinutte can you please check?

**Comment by Albinutte** [comment]
@kokosek can you please give a link and maybe a code?

**Comment by kokosek** [comment]
@Albinutte Sure.
The puzzle:
'std::string reflectString(std::string inputString) {

  std::string result = "";
  for (int i = 0; i < inputString.size(); i++) {
    int order = int(inputString[i]) - int('a'),
        reflectedCharCode = int('a') + 25 - order;
    result += int(reflectedCharCode);
  }

  return result;
}'
...and the correct version:
`std::string reflectString(std::string inputString) {

  std::string result = "";
  for (int i = 0; i < inputString.size(); i++) {
    int order = int(inputString[i]) - int('a'),
        reflectedCharCode = int('a') + 25 - order;
    result += reflectedCharCode;
  }

  return result;
}`

**Comment by kokosek** [comment]
Another one:
```
std::string mySubstringByLength(std::string inputString, int start, int length) {
  std::vector<char> result;
  for (int i = start; i <= start + length; i++) {
    result.push_back(inputString[i]);
  }
  return std::string(result.begin(), result.end());
}
```
...and correct version:
```
std::string mySubstringByLength(std::string inputString, int start, int length) {
  std::vector<char> result;
  for (int i = start; i < start + length; i++) {
    result.push_back(inputString[i]);
  }
  return std::string(result.begin(), result.end());
}
```

**Comment by Albinutte** [comment]
@tigransloyan yup, I managed to reproduce it [here](https://codefights.com/challenge/eiPeieHyk4ZpF9wod): changing 
`auto blah = mySubstringByLength(arg1, 0, 2);`
into
`auto blah = mySubstringByLength(arg1, 0, 3);`
causes `"Syntax Error"`.

The weird thing is it fails even on the first test, although its input is long enough.

**Comment by tigransloyan** [comment]
@eduard-piliposyan ^^

**Comment by eduard-piliposyan** [comment]
@Albinutte @tigransloyan the only issue here is "syntax error" phrase it should be "run time error" for some cases.
About Albina's question. We run all tests together, thus if you get `run time error/syntax error/time limit exceeded` you can't find out which test it was. It can be first one, or the last one, but verdict is one for all test cases.

**Comment by kokosek** [comment]
@eduard-piliposyan Yes, I agree that "run time error" would be ok. Still, "syntax error" is misleading.
